File: VonexDealer.Vue/ClientApp/models/HardwareOrder.js
export class hardwareOrder {
    constructor({

        hardwareOrderID = 0,
        orderID = 0,
        addressID = 0,
        hardwareOrderDetails = null,
        subTotal = 0.00,
        freight = 0.00,
        totalInitialPayment = 0.00,
        

    } = {}) {
            this.hardwareOrderID = hardwareOrderID;
            this.orderID = orderID;
            this.addressID = addressID;
            this.hardwareOrderDetails = hardwareOrderDetails;
            this.subTotal = subTotal;
            this.freight = freight;
            this.totalInitialPayment = totalInitialPayment;

    }
}

export function CreateHardwareOrder(data) {
    return Object.freeze(new hardwareOrder(data))
}

File: VonexDealer.API/Models/Order/HardwareOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace VonexDealer.API.Models.Order
{
    public class HardwareOrder
    {
        public Int64 HardwareOrderID { get; set; }
        public Int64 OrderID { get; set; }
        public Int64 AddressID { get; set; }
        public List<Hardware> HardwareOrderDetails { get; set; }

        public double SubTotal { get; set; }
        public double Freight { get; set; }
        public double TotalInitialPayment
        {
            get
            {
                return SubTotal + Freight;
            }
        }
    }
}
